classes/indoorFloor.tsx

```diff
@@ -39,59 +45,174 @@ class IndoorFloor {
         this.initialHeightPosition = newInitialHeightPosition;
     }
 
-    setImageWidth(newImageWidth: number){
+    setImageWidth (newImageWidth: number) {
         this.imageWidth = newImageWidth;
         const ratio = this.floorData.imageWidthPX / this.floorData.imageHeightPX;
-        this.imageHeight = newImageWidth*ratio;
+        this.imageHeight = newImageWidth * ratio;
     }
 
-    showFloorImage(){
-        return(
+    setFloorImageStyle():ImageStyle{
+        return{
+            width: this.imageWidth, height: this.imageHeight, position: "absolute", top: this.initialHeightPosition, left: this.initialWidthPosition, zIndex: 4
+        }
+    }
+    showFloorImage() {
+        return (
             <Image
                 resizeMode={'cover'}
-                style={{width: this.imageWidth, height: this.imageHeight, position: "absolute", top: this.initialHeightPosition, left: this.initialWidthPosition, zIndex: 4}}
+                style={this.setFloorImageStyle()}
                 source={this.floorData.floorImage}
             />
         )
     }
+    // This function has been commented out because it's out of use. It could be useful when updating the infloormap data in the future though
+    // Concerning the walkables paths
+
+    // getWalkwayTileColor(xIndex, yIndex) {
+    //     let color = 'red';
+    //     this.floorData.walkways.forEach(obj => {
+    //         if (xIndex >= obj.topLeft.x && xIndex <= obj.bottomRight.x && yIndex >= obj.topLeft.y && yIndex <= obj.bottomRight.y) {
+    //             color = 'blue';
+    //         }
+    //     })
+    //     if (this.floorData.entrance && xIndex === this.floorData.entrance.x && yIndex === this.floorData.entrance.y) {
+    //         color = 'white'
+    //     }
+    //     if (this.floorData.elevator && xIndex === this.floorData.elevator.x && yIndex === this.floorData.elevator.y) {
+    //         color = 'black'
+    //     }
+    //     if (this.floorData.bathrooms) {
+    //         this.floorData.bathrooms.forEach(bathroom => {
+    //             if (xIndex === bathroom.x && yIndex === bathroom.y) {
+    //                 color = 'yellow'
+    //             }
+    //         })
+    //     }
+    //     if (this.floorData.waterFountains) {
+    //         this.floorData.waterFountains.forEach(waterFountain => {
+    //             if (xIndex === waterFountain.x && yIndex === waterFountain.y) {
+    //                 color = 'purple'
+    //             }
+    //         })
+    //     }
+    //     if (this.floorData.classRooms) {
+    //         this.floorData.classRooms.forEach(classRoom => {
+    //             if (xIndex === classRoom.location.x && yIndex === classRoom.location.y) {
+    //                 color = 'orange'
+    //             }
+    //         })
+    //     }
+    //     return color;
+    // }
+
+// this function returns proper styles for the icons
+    setLogoStyle=( xPosition, yPosition, type): ImageStyle => {
```

Comment:
solid function, but as you see there are a few properties that are reused in every if statement. To avoid this repetition, set a styles variable at the top and define all those constants there. Ex:

```
let styles = {zIndex: 7, position: 'absolute', top: yPosition, left: xPosition}
if(toilet)
    styles['borderRadius'] = 25/2
    return styles
```